Assignments/Program.cs

```diff
@@ -1 +1,131 @@
+                rTestCases = new () { 0, 1, 4, 5, 9, 10, 47, 94, 100, 345, 400, 499, 500, 900, 998, 1000, 
+                                    1500, 1729, 1888, 3495, 4000, 3999 };
+      Console.WriteLine ("\x1B[4m" + "Numbers to Words:" + "\x1B[0m");
+      PrintResult (wTestCases, true);
+      Console.WriteLine ("\n\x1B[4m" + "Numbers to Romans:" + "\x1B[0m");
+      PrintResult (rTestCases, false);
+   }
+
+   /// <summary>Prints the Words or Roman numerals equivalent to the input number</summary>
+   public static void PrintResult (List<int> input, bool toWords) {
+      for (int i = 0; i < input.Count; i++) {
+         int n = input[i];
+         Console.Write ($"{i + 1,2}) {input[i],10} ==> ");
+         if (!toWords && n > 3999) {
+            Console.WriteLine ($"Invalid Input!");
+            continue;
+         }
+         Console.ForegroundColor = Green;
+         Console.WriteLine (toWords ? ToWords (input[i]) : ToRomans (input[i]));
+         Console.ResetColor ();
+      }
+   }
+
+   /// <summary>Returns the input number in words using State Machine</summary>
+   public static string ToWords (int num) {
+      if (num == 0) return "Zero";
+      string input = num.ToString ().PadLeft (9, '0'), 
+            output = "";
+      List<string> ones = new () { "", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten",
+      "Eleven", "Twelve", "Thirteen", "Fourteen", "Fifteen", "Sixteen", "Seventeen", "Eighteen", "Nineteen" },
+                   tens = new () { "", "Ten", "Twenty", "Thirty", "Forty", "Fifty", "Sixty", "Seventy", "Eighty", "Ninety" };
+
+      /*
+       This method works for a maximum of 9-digit number, and its process is explained below:
+      1. In the FIRST for-loop, input string of length 9 is segregated into 3 + 3 + 3 digits, each parsed into
+         3 digit integer, and made run into the second for-loop.
+         Eg: num = 123456, then input = "000123456" => part = { 000 (for i = 0); 123 (for i = 1); 456 (for i = 2) }
+      
+      2. In the SECOND for-loop, each single digits from 'part' stored in 'nthDigit' is taken out one by one and made 
+         run through the state machine.
+         Eg: part = 123 => nthDigit = { 3 (for n = 3); 2 (for n = 2); 1 (for n = 1) }
+             thirdDigit = 3 (3rd digit of part)
+
+      3. The state machine starts with the state of HUNDREDS as the maximum place value of any 3-digit number is 100.
+     
+      Overall, the first loop segregates the input string into three 3-digit numbers and supplies to the second loop,
+      which converts the supplied 3-digit number to equivalent words.  Finally all the words are added together and returned.
+       */
+
+      // FIRST for-loop:
+      for (int i = 0; i < 3; i++) {
+         int part = int.Parse (input.Substring (i * 3, 3));
+         State s = State.Hundreds;
+         // SECOND for-loop:
+         for (int n = 3; n > 0; n--) {

```

Comment:
As per your comments, for converting 1 to words, 000000001 first two iterations of this with just zeros is unnecessary. Can't this be fixed?